frontend/src/components/TruncatedText.tsx

```diff
@@ -6,21 +6,55 @@ type TruncatedTextProps = {
   content: string;
 } & React.HTMLProps<HTMLSpanElement>;
 
-const TruncatedText: React.FC<TruncatedTextProps> = ({ maxLines, content, ...rest }) => (
-  <Tooltip content={content}>
+const TruncatedText: React.FC<TruncatedTextProps> = ({ maxLines, content, ...props }) => {
+  const [outerElement, setOuterElement] = React.useState<HTMLElement | null>(null);
+  const [textElement, setTextElement] = React.useState<HTMLElement | null>(null);
+  const [isHover, setIsHover] = React.useState<boolean>(false);
+  const [isTruncated, setIsTruncated] = React.useState<boolean>(false);
+
+  const textRef = (ref: HTMLElement | null) => {
+    setTextElement(ref);
+  };
+
+  const outerRef = (ref: HTMLElement | null) => {
+    setOuterElement(ref);
+  };
+
+  React.useEffect(
+    () =>
+      setIsTruncated(
+        !!textElement &&
+          !!outerElement &&
+          isHover &&
+          textElement.offsetHeight > outerElement.offsetHeight,
+      ),
+    [textElement, outerElement, isHover],
+  );
+
+  const truncateBody = (
     <span
       style={{
         display: '-webkit-box',
-        overflow: 'hidden',
         WebkitBoxOrient: 'vertical',
-        WebkitLineClamp: maxLines,
         overflowWrap: 'anywhere',
+        overflow: 'hidden',
+        WebkitLineClamp: maxLines,
       }}
-      {...rest}
+      ref={outerRef}
+      onMouseEnter={() => setIsHover(true)}
+      onMouseLeave={() => setIsHover(false)}
```

Comment:
It is only necessary to determine at the time of mouse over whether the tooltip needs to be shown or not. No need to do a reset on mouse leave.
Furthermore, if we remove the `isHover` state, update the `isTruncated` state immediately.
```
  const updateTruncation = React.useCallback(() => {
    if (textRef.current && outerRef.current) {
      setIsTruncated(textRef.current.offsetHeight > outerRef.current.offsetHeight);
    }
  }, []);
```
```suggestion
      onMouseEnter={updateTruncation}
```